certification/pyxis/pyxis.go

```diff
@@ -126,6 +127,59 @@ func (p *pyxisClient) getImage(ctx context.Context, dockerImageDigest string) (*
 	return &data.Data[0], nil
 }
 
+// FindImagesByDigest uses an unauthenticated call to find_images() graphql function, and will
+// return a slice of CertImages. It accepts a slice of image digests. Additional fields
+// for return should be specified in ContainerImage
+func (p *pyxisClient) FindImagesByDigest(ctx context.Context, digests []string) ([]CertImage, error) {
+	// our graphQL query
+	var query struct {
+		FindImages struct {
+			ContainerImage []struct {
+				ID        graphql.String  `graphql:"_id"`
+				Certified graphql.Boolean `graphql:"certified"`
+			} `graphql:"data"`
+			Error struct {
+				Status graphql.Int    `graphql:"status"`
+				Detail graphql.String `graphql:"detail"`
+			} `graphql:"error"`
+			Total graphql.Int
+			Page  graphql.Int
+			// filter to make sure we get exact results
+		} `graphql:"find_images(filter: {docker_image_digest:in:$digests})"`
+	}
+
+	graphqlDigests := make([]graphql.String, 0, len(digests))
+	for _, digest := range digests {
+		graphqlDigests = append(graphqlDigests, graphql.String(digest))
+	}
+	// variables to feed to our graphql filter
+	variables := map[string]interface{}{
+		"digests": graphqlDigests,
+	}
+
+	// make our query
+	httpClient, ok := p.Client.(*http.Client)
+	if !ok {
+		return nil, fmt.Errorf("client could not be used as http.Client")
+	}
+	client := graphql.NewClient(p.getPyxisGraphqlUrl(), httpClient)
+
+	err := client.Query(ctx, &query, variables)
+	if err != nil {
+		return nil, fmt.Errorf("error while executing find_images query: %v", err)
+	}
+
+	images := make([]CertImage, 0, len(query.FindImages.ContainerImage))
+	for _, image := range query.FindImages.ContainerImage {
+		images = append(images, CertImage{
+			ID:        string(image.ID),
+			Certified: bool(image.Certified),
```

Comment:
I think it might be usefully to also add the `DockerImageDigest` since in our code the `ID` has no meaning. I'm thinking logging in the check, and things of that nature that it would be helpful for.

Reply:
Sure. I can add that.